src/Workspaces/CSharp/Portable/CodeStyle/PreferBracesPreference.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    /// <summary>
+    /// Defines the known values for <see cref="CSharpCodeStyleOptions.PreferBraces"/>.
+    /// </summary>
+    internal enum PreferBracesPreference
+    {
+        /// <summary>
+        /// Braces are allowed, but not preferred.
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// <para>Braces are always allowed, and generally preferred except in limited situations involving single-line
+        /// statements and expressions:</para>
+        ///
+        /// <list type="bullet">
+        /// <item><description>Braces may be omitted in the cases described for <see cref="Always"/>.</description></item>
+        /// <item><description>Braces may be omitted when the entire statement is placed on one line.</description></item>
+        /// <item><description>For a statement that contains one or more embedded statements, braces may be omitted when
+        /// every embedded statement fits on one line, and the part preceding the embedded statement is placed on one
+        /// line. If any embedded statement uses braces, braces are preferred for all embedded statments of the same
+        /// parent statement. For the purposes of evaluating this rule, if the embedded statement following an
+        /// <c>else</c> keyword is an if statement, the embedded statements of the nested if statement are treated as
+        /// children of the parent statement of the <c>else</c> keyword.</description></item>
+        /// </list>
+        /// </summary>
+        WhenMultiline,
```

Comment:
it bothers me this has the '1' value.

Reply:
Note: in general when we've had boolean->enum migration.  We keep 0/1 meaning false/true, and we've added the new values from 2 onwards.